aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move

```diff
@@ -51,120 +52,171 @@ module 0x1::aggregator_v2_test {
+        let vector_data = table::borrow_mut(data, outer);
+        if (vector::length(vector_data) == inner) {
+            vector::push_back(vector_data, option::none());
+        };
+
+        let option_data = vector::borrow_mut(vector_data, inner);
+
+        option::fill(option_data, e);
+    }
+
+    inline fun for_element_ref<Element: store, R>(account: &signer, i: u64, f: |&Element|R): R acquires AggregatorStore {
         let addr = signer::address_of(account);
-        let aggregators = &borrow_global<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = table::borrow(aggregators, i);
-        let actual = aggregator_v2::read(aggregator);
-        assert!(actual == expected, ENOT_EQUAL)
+        let data = &borrow_global<AggregatorStore<Element>>(addr).data;
+        let outer = i / 10;
+        let inner = i % 10;
+        let vector_data = table::borrow(data, outer);
+        let option_data = vector::borrow(vector_data, inner);
+        let value = option::borrow(option_data);
+
+        f(value)
     }
+    inline fun for_element_mut<Element: store, R>(account: &signer, i: u64, f: |&mut Element|R): R acquires AggregatorStore {
+        let addr = signer::address_of(account);
+        let data = &mut borrow_global_mut<AggregatorStore<Element>>(addr).data;
+        let outer = i / 10;
+        let inner = i % 10;
+        let vector_data = table::borrow_mut(data, outer);
+        let option_data = vector::borrow_mut(vector_data, inner);
+        let value = option::borrow_mut(option_data);
+        f(value)
+    }
+
 
     public entry fun new(account: &signer, i: u64, limit: u128) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &mut borrow_global_mut<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = aggregator_v2::create_aggregator(limit);
-        table::add(aggregators, i, aggregator);
+        insert<Aggregator<u128>>(account, i, aggregator_v2::create_aggregator(limit));
+    }
+
+    public fun call_read(account: &signer, i: u64): u128 acquires AggregatorStore {
+        for_element_ref<Aggregator<u128>, u128>(account, i, |aggregator| aggregator_v2::read(aggregator))
     }
 
     public entry fun try_add(account: &signer, i: u64, value: u128) acquires AggregatorStore {
-        let addr = signer::address_of(account);
-        let aggregators = &mut borrow_global_mut<AggregatorStore>(addr).aggregators_u128;
-        let aggregator = table::borrow_mut(aggregators, i);
-        aggregator_v2::try_add(aggregator, value);
+        for_element_mut<Aggregator<u128>, bool>(account, i, |aggregator| aggregator_v2::try_add(aggregator, value));
+    }
+
+    public entry fun add(account: &signer, i: u64, value: u128) acquires AggregatorStore {
+        for_element_mut<Aggregator<u128>, bool>(account, i, |aggregator| {
+            aggregator_v2::add(aggregator, value);
+            true
```

Comment:
The `add` function always returns true? Shouldn't the output depend on the output of `try_add`?